### PR TITLE
[14.0][FIX] hr_expense_cancel: Fix tests + Cancel move payments. 

### PR DESCRIPTION
**Pull request by victoralmau**
Fix tests + Cancel move payments (Since odoo/odoo@64b2580).

Please @ernestotejeda and @pedrobaeza can you review it? 

@Tecnativa TT33774

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Why this is needed? Remember to put always the whys on the commit message and pr description

**Comment by victoralmau** [comment]
Sorry, the PR was in draft until the definitive solution was added. It is already explained a bit and especially the commit that has caused the error in the tests.

**Review by ernestotejeda — COMMENTED** [review]
Code review 👍🏼

**Review by ernestotejeda — APPROVED** [review]

File: hr_expense_cancel/models/hr_expense.py [hunk 1]
```diff
@@ -153,7 +153,7 @@ def _remove_move_reconcile(self, payments, account_move):
 
     def _cancel_payments(self, payments):
         for rec in payments:
-            rec.move_id.with_context({"force_delete": True}).unlink()
+            rec.move_id.button_cancel()
```
Comment (pedrobaeza) [inline hr_expense_cancel/models/hr_expense.py:156]: Are these payments later reused?

Comment (victoralmau) [reply hr_expense_cancel/models/hr_expense.py:156]: I honestly don't know (I'd have to check it out).

Before the change of odoo with this code the payment was deleted but now it is not deleted and I sincerely believe that it is the right thing to do and that what should be done is to cancel the move (which also cancel the payment).

**Review by pedrobaeza — APPROVED** [review]
OK, let's keep it

**Comment by pedrobaeza** [comment]
/ocabot merge patch

Should this go to v15?

**Comment by victoralmau** [comment]
> 

Yes, this will also be necessary in v15 but the migration PR is not yet there.

**Comment by pedrobaeza** [comment]
OK, that's what I meant, if the module is already migrated or not. Check also possible PRs for advertising it.

**Pull request merged**